src/components/aws/tags/ChartComponent.js

```diff
@@ -0,0 +1,174 @@
+        </div>
+
+        <div className="inline-block pull-right">
+
+          {keys}
+
+          <div className="inline-block">
+            <IntervalNavigator
+              startDate={this.props.dates.startDate}
+              endDate={this.props.dates.endDate}
+              setDatesFunc={this.setDates}
+              interval={this.props.interval}
+              setIntervalFunc={this.setInterval}
+            />
+          </div>
+
+          {close}
+
+        </div>
+
+      </div>
+    );
+  }
+
+}
+
+Header.propTypes = {
+  id: PropTypes.string.isRequired,
+  accounts: PropTypes.arrayOf(PropTypes.object),
+  keys: PropTypes.object,
+  getKeys: PropTypes.func.isRequired,
+  tag: PropTypes.string.isRequired,
+  selectKey: PropTypes.func.isRequired,
+  values: PropTypes.object,
+  getValues: PropTypes.func.isRequired,
+  dates: PropTypes.object.isRequired,
+  setDates: PropTypes.func.isRequired,
+  interval: PropTypes.string.isRequired,
+  setInterval: PropTypes.func.isRequired,
+  close: PropTypes.func
+};
+
+class ChartComponent extends Component {
+
+  componentWillMount() {
+    this.props.getKeys(this.props.id, this.props.dates.startDate, this.props.dates.endDate);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.dates !== nextProps.dates || this.props.accounts !== nextProps.accounts) {
+      nextProps.getKeys(nextProps.id, nextProps.dates.startDate, nextProps.dates.endDate);
+      nextProps.selectKey(nextProps.id, "");
+    }
+    else if (this.props.keys !== nextProps.keys && nextProps.keys.status
+      && nextProps.keys.hasOwnProperty("values") && nextProps.keys.values.length)
+      nextProps.selectKey(nextProps.id, nextProps.keys.values[0]);
+    else if ((this.props.tag !== nextProps.tag && nextProps.tag !== ""))
+      nextProps.getValues(nextProps.id, nextProps.dates.startDate, nextProps.dates.endDate, nextProps.tag);
+  }
+
+  render() {
+    let error = null;
+
+    if (this.props.keys && this.props.keys.status) {
+      if (this.props.keys.hasOwnProperty("error"))
+        error = (<div className="alert alert-warning" role="alert">Data not available ({this.props.keys.error.message})</div>);
+      else if (this.props.keys.hasOwnProperty("values") && !this.props.keys.values.length)
+        error = (<div className="alert alert-warning" role="alert">Data not available (No keys available for this timerange)</div>);
+    }
+    if (this.props.values && this.props.keys.values && this.props.keys.hasOwnProperty("error"))
+      error = (<div className="alert alert-warning" role="alert">Data not available ({this.props.keys.error.message})</div>);
+
+    const chart = (error === null && this.props.values && this.props.values.status && this.props.values.hasOwnProperty("values") ? (
+      <PieChart
+        values={this.props.values.values}
+        legend
+        height={450}
+      />
+    ) : null);
+
+    return (
+      <div className="clearfix">
+        <Header {...this.props}/>
+        {error}
```

Comment:
This error message has no margins. It is stuck to the time selection component.
![screen shot 2018-09-06 at 15 07 38](https://user-images.githubusercontent.com/3480526/45159519-3d5f0800-b1e7-11e8-8a0d-40c6f523bcd0.png)